src/interdiff.c

```diff
@@ -98,23 +120,57 @@ static int ignore_components = 0;
 static int ignore_components_specified = 0;
 static int unzip = 0;
 static int no_revert_omitted = 0;
+static int use_colors = 0;
+static int color_option_specified = 0;
 static int debug = 0;
 
 static struct patlist *pat_drop_context = NULL;
 
 static struct file_list *files_done = NULL;
+
 static struct file_list *files_in_patch2 = NULL;
 static struct file_list *files_in_patch1 = NULL;
 
-/* checks whether file needs processing and sets context */
+/*
+ * Print colored output using a variadic format string.
+ */
+ static void
+ print_color (FILE *output_file, enum line_type type, const char *format, ...)
+ {
+	const char *color_start = "";
+	const char *color_end = "";
+	va_list args;
+
+	/* Only colorize if colors are enabled AND we're outputting to stdout */
+	if (use_colors && output_file == stdout) {
+		color_start = color_codes[type];
+		if (color_start[0] != '\0')
+			color_end = COLOR_RESET;
+	}
+
+	/* Print color start code */
+	if (color_start[0] != '\0')
```

Comment:
Change to `if (color_start)` after the change to initialize `color_start` to NULL.